### PR TITLE
LTE message parsing

### PR DESCRIPTION
**Pull request by martin-heusse**
Finds the size and copies to the LTE message from the diag structure. Up/down flag of RRC message looks better now.

### CODE REVIEW

**Comment by zecke** [comment]
Thank you! Which wireshark patch do you use to look at the result?

**Comment by martin-heusse** [comment]
Hi, 
I have a patched wireshark which does the job.
It's just an updated packet-gsmtap dissector that I want to push now

**Comment by zecke** [comment]
Great. Then please push it to the Wireshark Gerrit, share the link and I can check how my traces look with your version of the modified dissector! And thank you for adding upstream support  for LTE decoding!

**Comment by martin-heusse** [comment]
Here is the patch on gerrit
https://code.wireshark.org/review/#/c/22515/

Best regards,

M. Heusse

**Comment by E3V3A** [comment]
Also note: 
https://github.com/wireshark/wireshark/pull/12#issuecomment-313229603

File: diag_input.c [hunk 1]
```diff
@@ -242,30 +235,32 @@ struct radio_message * handle_4G(struct diag_packet *dp, unsigned len)
...
+			m->bb.arfcn[0] |= ARFCN_UPLINK;
 			break;
 		default:
 			// Unhandled
 			return NULL;
 		}
 		// verify len
-		payload_len = ((uint16_t)dp->data[9]) << 8 | dp->data[8];
-		if (payload_len > len - 15) {
+		payload_len = dp->data[12]-1;
```
Comment (zecke) [inline diag_input.c]: Are you sure that the length field is only 8bit? I would expect 16bit here?

Comment (martin-heusse) [reply diag_input.c]: This length does not reflect the actual size of the message anyway.
This is corrected in a subsequent commit

Comment (zecke) [reply diag_input.c]: But these commits are not in this pull request yet? I am not a LTE expert but I would expect QXDM to have a 16bit length field and LTE to have longer messages too. bb.data can be > 256 as well.

I can merge anyway but I would expect us to see truncated messages?

Comment (martin-heusse) [reply diag_input.c]: There are 4 commits in this pull request
https://github.com/moiji-mobile/diag-parser/pull/6/commits

Comment (zecke) [reply diag_input.c]: Yes. but I look at files changed which is "git diff base..4thcommit"? So payload_len = dp->data[12]-1 is the final one. Do you have an indication of why this field only needs to be 8bit and not 16bit?

Comment (martin-heusse) [reply diag_input.c]: with all 4 commits, it looks like this :
		/* payload_len = dp->data[12]-1; */
		payload_len=len-14;
		printf("payload_len: %d\n",payload_len);
		data = &dp->data[14];

Comment (martin-heusse) [reply diag_input.c]: I should probably remove the comment

Comment (zecke) [reply diag_input.c]: Okay. Not sure how I ended up on the patch. I thought I clicked PR->files changed.. Hmm. Could you remove the /* payload_len .. */ and then merge it? Thank you!

**Pull request closed without merging**